xbmc/filesystem/CurlFile.cpp

```diff
@@ -1918,6 +1910,25 @@ int CCurlFile::IoControl(EIoControl request, void* param)
   return -1;
 }
 
+const std::string CCurlFile::GetProperty(XFILE::FileProperty type, const std::string &name) const
+{
+  switch (type)
+  {
+  case FILE_PROPERTY_RESPONSE_PROTOCOL:
+    return m_state->m_httpheader.GetProtoLine();
+  case FILE_PROPERTY_RESPONSE_HEADER:
+    return m_state->m_httpheader.GetValue(name);
+  case FILE_PROPERTY_CONTENT_TYPE:
+    return m_state->m_httpheader.GetValue("content-type");
+  case FILE_PROPERTY_CONTENT_CHARSET:
+    return m_state->m_httpheader.GetCharset();
+  case FILE_PROPERTY_MIME_TYPE:
+    return m_state->m_httpheader.GetMimeType();
+  default:
+    return StringUtils::Empty;
```

Comment:
Why not simply `return "";`? `StringUtils::Empty` is mainly a trick if you have to return a reference to a string instead of an  instance. As this method is not called that often, I also cannot see any performance gains with `StringUtils::Empty`.